### PR TITLE
Separate speaker and addressee logic

### PR DESCRIPTION
**Pull request by msmolyak**
At your request I did not add tests or model tagger files to the project. I ran the code in my environment to make sure it still works. I will add unit tests when I pt ChangeGender into a separate project.

### CODE REVIEW

File: src/edu/stanford/nlp/international/russian/process/ChangeGenderConstants.java [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    static {
+        SIBILANTS_AND_VELARS.addAll(SIBILANTS);
+        SIBILANTS_AND_VELARS.addAll(VELARS);
+    }
+
+    final static Set<Character> SIBILANTS_FOR_NOUNS = Sets.toSet('ж', 'ч', 'ш', 'щ', 'ц');
+
+    final static Set<String> RELEVANT_PRONOUNS = Sets.toSet("я", "ты");
+
+    final static Set<String> RELEVANT_PERSONS = Sets.toSet("1", "2");
```
Comment (gcooper-isi) [inline src/edu/stanford/nlp/international/russian/process/ChangeGenderConstants.java:49]: No longer used

File: src/edu/stanford/nlp/international/russian/process/ChangeGenderConstants.java [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+    final static Set<Character> SIBILANTS_AND_VELARS = new HashSet<>();
+    static {
+        SIBILANTS_AND_VELARS.addAll(SIBILANTS);
+        SIBILANTS_AND_VELARS.addAll(VELARS);
+    }
+
+    final static Set<Character> SIBILANTS_FOR_NOUNS = Sets.toSet('ж', 'ч', 'ш', 'щ', 'ц');
+
+    final static Set<String> RELEVANT_PRONOUNS = Sets.toSet("я", "ты");
```
Comment (gcooper-isi) [inline src/edu/stanford/nlp/international/russian/process/ChangeGenderConstants.java:47]: No longer used

File: src/edu/stanford/nlp/international/russian/process/ChangeGender.java [hunk 3]
```diff
@@ -30,330 +29,184 @@
...
+     * the speaker and/or addressee genders.
+     * @param text text to process
+     * @param targetSpeakerGender target speaker gender (M, F or U - unchanged)
+     * @param targetAddresseeGender target addressee gender (M, F or U - unchanged)
+     * @param pipeline StanfordCoreNLP pipeline configured to parse Russian text
+     * @return text with genders adjusted
+     */
+    public static String adjustGenderForText(String text, TargetGender targetSpeakerGender, TargetGender targetAddresseeGender, StanfordCoreNLP pipeline) {
+        if (targetAddresseeGender == TargetGender.U && targetSpeakerGender == TargetGender.U) {
+            // If no gender change requested, return teh original text
```
Comment (gcooper-isi) [inline src/edu/stanford/nlp/international/russian/process/ChangeGender.java:96]: Typo: "teh" should be "the"

**Comment by gcooper-isi** [comment]
`ChangeGenderTest.java` needs to be fixed to refer to the constants in their new location.

File: src/edu/stanford/nlp/international/russian/process/ChangeGenderConstants.java (hunk 1 above)
Comment (msmolyak) [reply src/edu/stanford/nlp/international/russian/process/ChangeGenderConstants.java]: Removed

File: src/edu/stanford/nlp/international/russian/process/ChangeGenderConstants.java (hunk 2 above)
Comment (msmolyak) [reply src/edu/stanford/nlp/international/russian/process/ChangeGenderConstants.java]: Removed

File: src/edu/stanford/nlp/international/russian/process/ChangeGender.java (hunk 3 above)
Comment (msmolyak) [reply src/edu/stanford/nlp/international/russian/process/ChangeGender.java]: Fixed

**Comment by msmolyak** [comment]
All fixed

**Pull request merged**